internal/resources/metal/vrf_bgp_dynamic_neighbor/resource.go

```diff
@@ -0,0 +1,174 @@
+package vrfbgpdynamicneighbor
+
+import (
+	"context"
+
+	"github.com/equinix/equinix-sdk-go/services/metalv1"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+var (
+	bgpNeighborIncludes = []string{"metal_gateway"}
+	// `created_by` is specified as a UserLimited.  To avoid an error
+	// due to missing UserLimited.id field, have to either exclude
+	// or include `created_by`
+	bgpNeighborExcludes = []string{"created_by", "ip_reservation"}
+)
+
+type Resource struct {
+	framework.BaseResource
+	framework.WithTimeouts
+}
+
+func NewResource() resource.Resource {
+	r := Resource{
+		BaseResource: framework.NewBaseResource(
+			framework.BaseResourceConfig{
+				Name: "equinix_metal_vrf_bgp_dynamic_neighbor",
+			},
+		),
+	}
+
+	return &r
+}
+
+func (r *Resource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	s := resourceSchema(ctx)
+	if s.Blocks == nil {
+		s.Blocks = make(map[string]schema.Block)
+	}
+
+	resp.Schema = s
+}
+
+func (r *Resource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var plan Model
+	response.Diagnostics.Append(request.Config.Get(ctx, &plan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	createRequest := metalv1.BgpDynamicNeighborCreateInput{
+		BgpNeighborRange: plan.Range.ValueString(),
+		BgpNeighborAsn:   plan.ASN.ValueInt64(),
+	}
+
+	response.Diagnostics.Append(getPlanTags(ctx, plan, &createRequest.Tags)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.CreateBgpDynamicNeighbor(ctx, plan.GatewayID.ValueString()).
+		BgpDynamicNeighborCreateInput(createRequest).
+		Exclude(bgpNeighborExcludes).
+		Include(bgpNeighborIncludes).
+		Execute()
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error creating VRF BGP dynamic neighbor range",
+			"Could not create VRF BGP dynamic neighbor range: "+err.Error(),
+		)
+	}
+
+	// Parse API response into the Terraform state
+	response.Diagnostics.Append(plan.parse(ctx, neighbor)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &plan)...)
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var data Model
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.BgpDynamicNeighborsIdGet(ctx, data.ID.ValueString()).
```

Comment:
Do we need to wait for a particular status here?  If so, do we wait for `active` or `ready`?  I'm assuming `pending` is the initial state but I don't see a description in the spec.